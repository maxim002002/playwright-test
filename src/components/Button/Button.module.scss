$primary-color: #3498db;
$white-color: #fff;
$button-background: #ecf0f1;
$button-border: #a5a5a5;
$button-text: #333;
$button-background-transparent: rgba(0, 0, 0, 0.06);

.button {
  display: inline-block;
  font-size: 16px;
  padding: 10px 20px;
  cursor: pointer;
  border-radius: 12px;
  transition:
    background-color 0.3s ease,
    border-color 0.3s ease,
    transform 0.1s ease,
    color 0.3s ease;
  min-width: 120px;
  user-select: none;
}
.button:disabled {
  opacity: 0.5;
  &:hover {
    background-color: darken($primary-color, 10%);
  }
}

.primary {
  background-color: $primary-color;
  color: $white-color;
  border: 1px solid $primary-color;
  &:hover {
    background-color: darken($primary-color, 10%);
  }
}

.default {
  background-color: $button-background;
  color: $button-text;
  border: 1px solid lighten($button-border, 20%);
  &:hover {
    border-color: $primary-color;
    color: $primary-color;
  }
}

.dashed {
  background-color: transparent;
  color: $button-text;
  border: 1px dashed $button-border;
  &:hover {
    border-color: $button-text;
    border-color: $primary-color;
    color: $primary-color;
  }
}

.text {
  background-color: transparent;
  color: $primary-color;
  border: none;
  &:hover {
    background-color: $button-background-transparent;
  }
}

.link {
  background-color: transparent;
  color: $primary-color;
  border: none;
  text-decoration: underline;
  &:hover {
    text-decoration: none;
    color: darken($primary-color, 10%);
  }
}
