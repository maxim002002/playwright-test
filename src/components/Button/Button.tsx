// @ts-ignore
import { FC } from "react";
import classnames from "classnames";
import styles from "./Button.module.scss";

export interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  name: "primary" | "default" | "dashed" | "text" | "link";
  type?: "submit" | "button";
  onClick?: VoidFunction;
  className?: string;
  disabled?: boolean;
  id?: string;
  dataTestId?: string;
}

export const Button: FC<IButton> = ({
  onClick,
  text,
  name,
  type,
  className,
  disabled,
  id,
  dataTestId,
}) => {
  return (
    <button
      className={classnames(styles.button, styles[name], className)}
      onClick={onClick}
      disabled={disabled}
      type={type}
      id={id}
      data-testid={dataTestId}
    >
      {text}
    </button>
  );
};
