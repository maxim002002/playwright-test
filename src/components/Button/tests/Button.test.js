// - Написать unit-тест
// - Написать snapshot на jest

import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "../Button";

describe("Button test cases", () => {
  test("renders button with text", () => {
    render(<Button text="Click me" name="default" />);
    const buttonElement = screen.getByText("Click me");
    expect(buttonElement).toBeInTheDocument();
  });

  test("handles onClick event", () => {
    const onClickMock = jest.fn();
    render(<Button text="Click me" name="default" onClick={onClickMock} />);
    const buttonElement = screen.getByText("Click me");
    fireEvent.click(buttonElement);
    expect(onClickMock).toHaveBeenCalled();
  });

  test("applies custom className", () => {
    render(<Button text="Click me" name="default" className="custom-class" />);
    const buttonElement = screen.getByText("Click me");
    expect(buttonElement).toHaveClass("custom-class");
  });

  test("renders disabled button", () => {
    render(<Button text="Click me" name="default" disabled />);
    const buttonElement = screen.getByText("Click me");
    expect(buttonElement).toBeDisabled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Button text="Click me" name="default" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
