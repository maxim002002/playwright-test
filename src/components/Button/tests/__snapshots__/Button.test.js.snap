// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button default"
  >
    Click me
  </button>
</DocumentFragment>
`;
